### PR TITLE
Beleroy/opencv site

### PR DESCRIPTION
**Pull request by bleroy**
## Purpose

Add an OpenCV image processing skill and associated tooling.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code
* Follow readme instructions
```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

**Comment by bleroy** [comment]
@mattmsft You'll also want to update the top-level solution and readme, see Sorin's PR for reference.

File: Vision/OpenCV/__app__/utilities/opencvskill.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    
+    def __add_debug_step(self, debug_step):
+        if (self.debug_mode):
+            self.debug_state.append(debug_step)
+
+    def get_output_debug_steps(self):
+        result = []
+        for debug_step in self.debug_state:
+            is_success, result_image = cv2.imencode(".png", debug_step["image"])
+            result_debug_step = { "image": { "$type": "file", "name": "output.png", "contentType": "image/png", "data": webapiskill.encode_image_bytes(result_image) } }
```
Comment (bleroy) [inline Vision/OpenCV/__app__/utilities/opencvskill.py:19]: Do you want to make some of those parameters configurable through env variables or parameters?

File: Vision/OpenCV/__app__/HttpTrigger/main.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    cv_skill = opencvskill(debug_override=True)
+    image_cv = cv_skill.imdecode(image_np, cv2.IMREAD_COLOR)
+    # Converting the image to grayscale.
+    gray = cv_skill.cvtColor(image_cv, cv2.COLOR_BGR2GRAY)
+    # Smoothing without removing edges.
+    gray_filtered = cv_skill.bilateralFilter(gray, 7, 50, 50)
+    # Applying the canny filter
+    edges_filtered = cv_skill.Canny(gray_filtered, 60, 120)
+
+    output_debug_steps = cv_skill.get_output_debug_steps()
```
Comment (bleroy) [inline Vision/OpenCV/__app__/HttpTrigger/main.py:32]: Eventually (not necessarily in this PR, later), do we want to introduce some kind of lightweight declarative way to specify your pipeline steps?

**Comment by amitkalay** [comment]
gonna close this one out since it's 4 years old.

**Pull request closed without merging**